File: 334. Increasing Triplet Subsequence.py
class Solution(object):
    def increasingTriplet(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        if (not nums) or len(nums) < 3:
            return False
        ret = 1
        x = nums[0]
        y = float('inf')
        for i in nums:
            if i < x:
                x = i
            elif i > x and i < y:
                y = i
            if i > y:
                return True
        return False
            
            
            
        

File: 334. Increasing Triplet Subsequence.go
func increasingTriplet(nums []int) bool {
	n := len(nums)
	if n < 3 {
		return false
	}
	maxP := 0
	for i := 0; i < n; i++ {
		j := maxP
		for j >= 0 && nums[i] <= nums[j] {
			j--
		}
		nums[j+1] = nums[i]
		if j+1 > maxP {
			maxP = j + 1
			if maxP == 2 {
				return true
			}
		}
	}
	return false
}
